cfme/rest/gen_data.py

```diff
@@ -223,25 +223,62 @@ def _finished():
     return vm_name
 
 
-def service_templates_ui(request, rest_api, service_dialog=None, service_catalog=None, num=4):
+def service_templates_ui(request, rest_api, service_dialog=None, service_catalog=None,
+        a_provider=None, vmdb=None, num=4):
     if not service_dialog:
         service_dialog = dialog()
     if not service_catalog:
         service_catalog = service_catalog_obj(request, rest_api)
 
+    catalog_item_type = 'Generic'
+    provisioning_args = {}
+
     catalog_items = []
     new_names = []
+    vm_names = []
     for _ in range(num):
+        if a_provider:
+            template, host, datastore, vlan, catalog_item_type = map(
+                a_provider.data.get('provisioning').get,
+                ('template', 'host', 'datastore', 'vlan', 'catalog_item_type'))
+
+            vm_name = 'test_rest_{}'.format(fauxfactory.gen_alphanumeric())
+            vm_names.append(vm_name)
+            provisioning_data = {
+                'vm_name': vm_name,
+                'host_name': {'name': [host]},
+                'datastore_name': {'name': [datastore]}
+            }
+
+            if a_provider.type == 'rhevm':
```

Comment:
**REQUIRED**
I would recommend to not compare strings. Please use `provider.one_of()` method